native/src/components/MapView.tsx

```diff
@@ -46,14 +50,28 @@ const MapView = ({ boundingBox, featureCollection }: MapViewPropsType): ReactEle
     zoomLevel: defaultViewportConfig.zoom,
     bounds
   }
+
+  const updateLocationPermission = useCallback((locationPermission: boolean) => {
+    setLocationPermission(locationPermission)
+    setFollowUserLocation(false)
+    if (locationPermission) {
+      setFollowUserLocation(true)
+    }
```

Comment:
```suggestion
    setFollowUserLocation(locationPermission)
```

Reply:
Isn't the `followUserLocation` always the same as `locationPermission`

Reply:
I think not. Sometimes you want to zoom around the map, but the UserLocation should stay visible - thus locationPermission. And if you want to center the camera around the curent position again, you press the button, which will reset the followUserLocation variable.

Reply:
Yeah right, but with the current implementation it is always the same, right?

Reply:
it may be. But to reset the camera you have to reset `followUserLocation`, which would lead to a flickering location indicator

Reply:
The suggestion still makes sense though I think :)

Reply:
yes. But with the previous setup it wouldn't have worked sadly.  I removed the separate LocationButtonComponent, and moved the permission logic to the mapview. Even though this blows up the map view, I think it becomes clearer, that `followUserLocation` is also changed by the mapview itself now and can influence the camera